ivy/functional/ivy/device.py

```diff
@@ -6,14 +6,20 @@
 import abc
 import math
 import psutil
-import pynvml
+
 import types
 from typing import Type, Optional, Tuple
 
 # noinspection PyUnresolvedReferences
 try:
-    pynvml.nvmlInit()
-except pynvml.NVMLError:
+    import pynvml
+
+    try:
+        pynvml.nvmlInit()
+    except pynvml.NVMLError:
+        pass
+except ImportError:
+    # nvidia-ml-py (pynvml) is not installed in CPU Dockerfile.
```

Comment:
Could you add a warning or throw an import error when GPU is present but pynvml is not installed? I feel it's better to show import warnings during initialization, than during actual work.